src/features/postsSlice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+export type PostsState = {
+  posts: Post[];
+  loading: boolean;
+  error: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loading: false,
+  error: false,
+};
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  (userId: number) => getUserPosts(userId),
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      return {
+        ...state,
+        posts: [],
+      };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchPosts.pending, state => {
+      return { ...state, loading: false };
+    });
+    builder.addCase(
+      fetchPosts.fulfilled,
+      (state, action: PayloadAction<Post[]>) => {
+        return {
+          ...state,
+          posts: action.payload,
+          loading: true,
+        };
+      },
+    );
+    builder.addCase(fetchPosts.rejected, state => {
+      return {
+        ...state,
+        loading: true,
```

Comment:
There is a mistake in the 'rejected' case. The 'loading' state should be set to 'false' when the fetch request is rejected, indicating that the loading process has stopped.